pyswarms/backend/handlers.py

```diff
@@ -0,0 +1,353 @@
+
+        By passing the handler, the :func:`compute_position()` functions now has
+        the ability to reset the particles by calling the :code:`BoundaryHandler`
+        inside.
+
+        Attributes
+        ----------
+        strategy : str
+            The strategy to use. To see all available strategies,
+            call :code:`BoundaryHandler.strategies`
+        """
+        self.strategy = strategy
+        self.strategies = self.__get_all_strategies()
+        self.rep = Reporter(logger=logging.getLogger(__name__))
+        self.memory = None
+
+    def __call__(self, position, bounds, **kwargs):
+        """Apply the selected strategy to the position-matrix given the bounds
+
+        Parameters
+        ----------
+        position : np.ndarray
+            The swarm position to be handled
+        bounds : tuple of :code:`np.ndarray` or list
+            a tuple of size 2 where the first entry is the minimum bound while
+            the second entry is the maximum bound. Each array must be of shape
+            :code:`(dimensions,)`
+        kwargs : dict
+
+        Returns
+        -------
+        numpy.ndarray
+            the adjusted positions of the swarm
+        """
+        # Combine `position` and `bounds` with extra keyword args
+        kwargs_ = self.__merge_dicts(
+            {"position": position, "bounds": bounds}, kwargs
+        )
+
+        try:
+            new_position = self.strategies[self.strategy](**kwargs_)
+        except KeyError:
+            message = "Unrecognized strategy: {}. Choose one among: " + str(
+                [strat for strat in self.strategies.keys()]
+            )
+            self.rep.log.exception(message.format(self.strategy))
+            raise
+        else:
+            return new_position
+
+    def __merge_dicts(self, *dict_args):
+        """Backward-compatible helper method to combine two dicts"""
+        result = {}
+        for dictionary in dict_args:
+            result.update(dictionary)
+        return result
+
+    def __out_of_bounds(self, position, bounds, velocity=None):
+        """Helper method to find indices of out-of-bound positions
+
+        This method finds the indices of the particles that are out-of-bound
+        if a velocity is specified it returns the indices of the particles that
+        will be out-of-bounds after the velocity is applied.
+        """
+        if velocity is not None:
+            position += velocity
+        lb, ub = bounds
+        greater_than_bound = np.nonzero(position > ub)
+        lower_than_bound = np.nonzero(position < lb)
+        return (lower_than_bound, greater_than_bound)
+
+    def __get_all_strategies(self):
+        """Helper method to automatically generate a dict of strategies"""
+        return {
+            k: v
+            for k, v in inspect.getmembers(self, predicate=inspect.isroutine)
+            if not k.startswith(("__", "_"))
+        }
+
+    def nearest(self, **k):
```

Comment:
Hi @whzup  

I realized that given a method:

```python
def nearest(pos, bounds):
    # do something
```

We can still pass a `**kwargs` argument such as `nearest(**kwargs)` as long as the dictionary is well-built. Can you check that one here? I think explicitly stating `posiiton` and `bounds` is better than just having an implicit `**kwargs`. There might be type errors in some special cases though but I think it would look better instead of indexing the kwargs dictionary for every parameter